pkg/placement/membership.go

```diff
@@ -15,32 +16,139 @@ import (
 const (
 	commandRetryInterval = 20 * time.Millisecond
 	commandRetryMaxCount = 3
+
+	barrierWriteTimeout = 2 * time.Minute
 )
 
-// MembershipChangeWorker is the worker to change the state of membership
-// and update the consistent hashing tables for actors.
-func (p *Service) MembershipChangeWorker() {
-	faultyHostDetectTimer := time.NewTicker(faultyHostDetectInterval)
-	disseminateTimer := time.NewTicker(disseminateTimerInterval)
+// MonitorLeadership is used to monitor if we acquire or lose our role
+// as the leader in the Raft cluster. There is some work the leader is
+// expected to do, so we must react to changes
+//
+// reference: https://github.com/hashicorp/consul/blob/master/agent/consul/leader.go
+func (p *Service) MonitorLeadership() {
+	var weAreLeaderCh chan struct{}
+	var leaderLoop sync.WaitGroup
+
+	leaderCh := p.raftNode.Raft().LeaderCh()
+
+	for {
+		select {
+		case isLeader := <-leaderCh:
+			switch {
```

Comment:
Will we anticipate more cases in this switch? Otherwise, change to `if-else`?

Reply:
yes will do that.